src/realm/util/serializer.cpp

```diff
@@ -43,18 +43,18 @@ static constexpr long date_to_julian(int y, int m, int d)
 
 static void julian_to_date(int jd, int* y, int* m, int* d)
 {
-    int L = jd + 68569;
-    int n = (4 * L) / 146097;
-    int i, j;
+    uint64_t L = jd + 68569;
+    uint64_t n = (4 * L) / 146097;
```

Comment:
This was undefined behavior where in some tests that log timestamps this multiplication could overflow an `int`. Switching to uint64_t's where overflowing is defined behavior.